ext/standard/basic_functions.c

```diff
@@ -1701,6 +1701,12 @@ PHPAPI void php_call_shutdown_functions(void) /* {{{ */
 			zend_hash_apply(BG(user_shutdown_function_names), user_shutdown_function_call);
 		} zend_end_try();
 	}
+
+	/* After this point it's no longer safe to call user exception handlers, so remove it. */
+	if (Z_TYPE(EG(user_exception_handler)) != IS_UNDEF) {
+		zval_ptr_dtor(&EG(user_exception_handler));
+		ZVAL_UNDEF(&EG(user_exception_handler));
+	}
```

Comment:
I'm not sure why it's no longer safe.
Anyway, I would move this into ``php_request_shutdown()`` to keep the shut-down sequence logic in a single place.

Reply:
Thanks for the review! The wording may be improved. It's a reference to what you wrote here:

> but note that behaviour of exception handlers is limited by each step of the shut-down sequence. (e.g. output buffers may be flushed, DB connections closed, objects destroyed, global and static variables unset)

Would you prefer if Exceptions were also called for dtors, and maybe output buffering handlers? Everything after seems more unlikely to throw.

Reply:
> Would you prefer if Exceptions were also called for dtors, and maybe output buffering handlers? Everything after seems more unlikely to throw.

I don't care a lot about the actual behaviour.
I just like to make the changes to be documented, to avoid the future changes in the shut-down sequence.

Reply:
@dstogov I actually had a look at this again, and noticed that error handlers *do* get triggered in all of these stages. To stay consistent, it makes more sense to adapt exception handlers to behave the same. I added tests for each of your bullet points [here](https://github.com/php/php-src/pull/10905#issuecomment-1485754288).